code/modules/mob/living/simple_animal/hostile/megafauna/onestarboss.dm

```diff
@@ -0,0 +1,178 @@
+	icon_living = "onestar_boss_unpowered"
+	icon_dead = "onestar_boss_wrecked"
+	pixel_x = -16
+	ranged = TRUE
+
+	health = 1700
+	maxHealth = 1700
+	break_stuff_probability = 95
+	stop_automated_movement = 1
+
+	melee_damage_lower = 10
+	melee_damage_upper = 20
+	megafauna_min_cooldown = 30
+	megafauna_max_cooldown = 60
+
+/mob/living/simple_animal/hostile/megafauna/one_star/death()
+	..()
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/LoseTarget()
+	stance = HOSTILE_STANCE_IDLE
+	target_mob = null
+	icon_state = initial(icon_state)
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/LostTarget()
+	stance = HOSTILE_STANCE_IDLE
+	icon_state = initial(icon_state)
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/FindTarget()
+
+	var/atom/T = null
+	var/valueofview = 0
+
+	if(src.loc:luminosity > 0)
+		valueofview = 10
+	else
+		valueofview = 4
+
+	for(var/atom/A in ListTargets(valueofview))
+
+		if(A == src)
+			continue
+
+		var/atom/F = Found(A)
+		if(F)
+			T = F
+			break
+
+		if(isliving(A))
+			var/mob/living/L = A
+			if(L.faction == src.faction && !attack_same)
+				continue
+			else if(L in friends)
+				continue
+			else
+				if(!SA_attackable(L))
+					stance = HOSTILE_STANCE_ATTACK
+					T = L
+					break
+
+		else if(istype(A, /obj/mecha)) // Our line of sight stuff was already done in ListTargets().
+			var/obj/mecha/M = A
+			if (M.occupant)
+				stance = HOSTILE_STANCE_ATTACK
+				T = M
+				break
+
+		if(istype(A, /obj/machinery/bot))
+			var/obj/machinery/bot/B = A
+			if (B.health > 0)
+				stance = HOSTILE_STANCE_ATTACK
+				T = B
+				break
+	if(T != null)
+		icon_state = "onestar_boss"
+	else
+		icon_state = initial(icon_state)
+	return T
+
+
+/mob/living/simple_animal/hostile/megafauna/one_star/Life()
+
+	. = ..()
+	if(!.)
+		walk(src, 0)
+		return 0
+	if(client)
+		return 0
+
+	if(!stat)
+		switch(stance)
+			if(HOSTILE_STANCE_IDLE)
+				target_mob = FindTarget()
+
+			if(HOSTILE_STANCE_ATTACK)
+				if(destroy_surroundings)
+					DestroySurroundings()
+				MoveToTarget()
+
+			if(HOSTILE_STANCE_ATTACKING)
+				if(destroy_surroundings)
+					DestroySurroundings()
+				AttackTarget()
+
+/mob/living/simple_animal/hostile/megafauna/one_star/AttackingTarget()
+	if(!Adjacent(target_mob))
+		return
+	if(isliving(target_mob))
+		var/mob/living/L = target_mob
+		L.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+		L.electrocute_act(10, src, safety = TRUE) //safety = TRUE means we don't check gloves...
+		return L
+	if(istype(target_mob,/obj/mecha))
+		var/obj/mecha/M = target_mob
+		M.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+		return M
+	if(istype(target_mob,/obj/machinery/bot))
+		var/obj/machinery/bot/B = target_mob
+		B.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+		return B
+
+/mob/living/simple_animal/hostile/megafauna/one_star/shoot_projectile(turf/marker, set_angle)
```

Comment:
What's the `set_angle` for? It's not used in the proc except for failing proc if it's not present.